templates/terraform/encoders/data_catalog_tag.go.erb

```diff
@@ -0,0 +1,17 @@
+       if obj["fields"] != nil {
+           // isEmptyValue() does not work for a boolean as it shows 
+           // false when it is 'empty'. Filter boolValue here based on 
+           // the rule api does not take more than 1 'value'
+                fields := obj["fields"].(map[string]interface{})
+                for _, elements := range fields {
+                        values := elements.(map[string]interface{})
+                        if len(values) > 1 {
```

Comment:
This seems to remove user-set boolValues, and even ones that are set to True? Even though we are enacting the API's rules, we should either validate during plan or let the user's bad configuration return an error, not change their values underneath the hood.

I think this encoder solves the problem of sending too many boolValue's, but checking if the user has set the field using `d.GetOkExists()` would do better here to make sure we are carrying out the user's intent. We can check if the value is set, and if not, delete it. How does that sound?

Reply:
I see.

Reply:
Actually backtracking on this: we can't use `d.GetOkExists()` because `fields` is a set and cannot be indexed. Don't really know any other way of tell the difference between user set/unset. Do you @ndmckinley ?

We stick with this original approach and only remove if the bool_value is false? Seems slightly better.

Reply:
Nothing sticks out to me.  Maybe we should pause this for a day or two and think on it, let it percolate, bring it up in the next Monday meeting if we haven't thought of anything clever by then.  What do you think?

Reply:
agreed!

Reply:
Does the API only except 1 value for `fields` ? If so we can use One-of validation to make sure the user isn't setting multiple values. Then we are free to delete the extra boolean values in this encoder

Reply:
Yes, the API only take 1 value for each fields. But there is no way to distinguish between `empty` and `false` for bool, along with other `value` types.